src/main/java/org/opensearch/neuralsearch/executors/HybridQueryExecutorCollector.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.executors;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Synchronized;
+
+import java.util.Optional;
+import java.util.function.Function;
+
+/**
+ * {@link HybridQueryExecutorCollector} is a generic Collector used by Hybrid Search Query during
+ * Query phase to parallelize sub query's action to improve latency
+ */
+@RequiredArgsConstructor(staticName = "newCollector")
+public final class HybridQueryExecutorCollector<I, R> {
```

Comment:
Did you explore other names for this abstraction? I have doubt about the 'Collector` suffix in the `HybridQueryExecutorCollector` name, it sounds similar to Lucene [Collector](https://lucene.apache.org/core/9_0_0/core/org/apache/lucene/search/Collector.html) which is in use for hybrid query, but your class is different. So to avoid ambiguity I would rename this interface and related classes with something without "Collector". Few possible examples for name prefix: HybridQueryExecutorJob or HybridQueryExecutorTask.

Reply:
I choose this suffix so that it follows Lucene's Collector/Manager pattern. This suffix is commonly used by every feature that wants to convey Manager/Entity pattern where collector is an entity which will be used only to collect the result .

Reply:
Understood, just in Lucene and thus OpenSearch Collector and CollectorManager is mainly used in context of collecting doc scores. Maybe I was looking into that doc collector code for too long :) I would change the class name, but I'm leaving the final decision to you.